build-tools/packages/build-cli/src/library/git.ts

```diff
@@ -165,9 +165,11 @@ export class Repository implements GitContext {
 	public async getShaForBranch(branch: string, remote?: string): Promise<string> {
 		const refspec =
 			remote === undefined ? `refs/heads/${branch}` : `refs/remotes/${remote}/${branch}`;
+		// result is a string of the form '64adcdba56deb16e0641c91ca825401a9f7a01f9 refs/heads/release/client/2.23'
```

Comment:
I noticed this code has not changed in 2 years, and `git show-ref` also doesn't seem to have changed its output in the 2.x train of git... have we really been that lucky with this somehow still working?

If we only care about the sha, I'd suggest we use `git show-ref --hash`, and that we also validate that the result contains a single line to be completely sure the sha is actually for the thing we intend.

Reply:
I can;t really explain it other than it's probably related to the changes in #22695. Maybe we did get lucky.